hera_qm/omnical_metrics.py

```diff
@@ -1,14 +1,872 @@
+            chisq_std_cut is like a "sigma" threshold for each antenna's
+            chisq variability.
+        """
+        # assign fc filename
+        self.firstcal_file = firstcal_file
+
+        # select freq channels
+        if cut_edges is True:
+            self.band = np.arange(Ncut, self.Nfreqs - Ncut)
+        else:
+            self.band = np.arange(self.Nfreqs)
+
+        # get chisq metrics
+        (tot_chisq, chisq_ant_avg, chisq_ant_std, chisq_ant_std_loc,
+         chisq_ant_std_scale, chisq_bad_ants) = self.chisq_metric(self.chisq, chisq_std_cut=chisq_std_cut)
+
+        if firstcal_file is not None:
+            # load fc gain solutions
+            self.firstcal_delays, self.firstcal_gains = load_firstcal_gains(firstcal_file)
+
+            # get gain difference between omnical and firstcal gain solutions
+            self.gain_diff = self.omni_gains * self.firstcal_gains.conj()
+
+            # run phs FT metric
+            (ant_phs_noise, tot_phs_noise, phs_noise_bad_ants,
+            phs_noise_good_sol) = self.phs_FT_metric(np.angle(self.gain_diff),
+                                                     phs_noise_cut=phs_noise_cut)
+
+            # run phs std metric
+            (ant_phs_std, tot_phs_std, phs_std_bad_ants,
+            phs_std_good_sol) = self.phs_std_metric(np.angle(self.gain_diff),
+                                                    phs_std_cut=phs_std_cut)
+
+        # initialize metrics
+        metrics = OrderedDict()
+
+        metrics['tot_chisq'] = tot_chisq
+        metrics['chisq_ant_avg'] = chisq_ant_avg
+        metrics['chisq_ant_std'] = chisq_ant_std
+        metrics['chisq_ant_std_loc'] = chisq_ant_std_loc
+        metrics['chisq_ant_std_scale'] = chisq_ant_std_scale
+        metrics['chisq_bad_ants'] = chisq_bad_ants
+        metrics['chisq_std_cut'] = chisq_std_cut
+
+        metrics['freqs'] = self.freqs
+        metrics['Nfreqs'] = self.Nfreqs
+        metrics['cut_edges'] = cut_edges
+        metrics['Ncut'] = Ncut
+        metrics['band'] = self.band
+        metrics['ant_array'] = self.ant_array
+        metrics['jones'] = self.jones
+        metrics['pol'] = self.pol
+        metrics['times'] = self.times
+        metrics['Ntimes'] = self.Ntimes
+        metrics['Nants'] = self.Nants
+
+        metrics['version'] = self.version_str
+        metrics['history'] = self.history
+        metrics['filename'] = self.filename
+        metrics['filestem'] = self.filestem
+        metrics['filedir'] = self.filedir
+
+        metrics['ant_phs_noise'] = None
+        metrics['tot_phs_noise'] = None
+        metrics['phs_noise_bad_ants'] = None
+        metrics['phs_noise_good_sol'] = None
+        metrics['phs_noise_cut'] = None
+        metrics['ant_phs_std'] = None
+        metrics['tot_phs_std'] = None
+        metrics['phs_std_bad_ants'] = None
+        metrics['phs_std_good_sol'] = None
+        metrics['phs_std_cut'] = None
```

Comment:
I just realized these `None`s may also lead to trouble with the M&C because the column requires floats. We could either come up with an obvious "this wasn't actually run" value (-1?), or in metrics2mc we can trim them out if their value is None.

Reply:
ok, I put in some if statements to trim them out in `metrics2mc` if they are None